### PR TITLE
MOB-1845 Use native alert to open external link instead os custom snack bar

### PR DESCRIPTION
**Pull request by lucaschifino**
[MOB-1845](https://ecosia.atlassian.net/browse/MOB-1845)

## Context
As suggested in the ticket, we should use the native alert when opening an external link.

## Approach
Update the already existing method and use `UIAlertController` instead of `TimerSnackBar`. Noticed this is only used for sms, mail, app store and as a default alert in case no other url type is found.

## Other
Also noticed that the `isStoreURL` method is not correctly identifying app store links anymore, I believe it to be because of the usage of `itunes.apple.com` instead of `https://apps.apple.com/`. Decided in the end not to change it since this is still the case in the [orginal ff repo](https://github.com/mozilla-mobile/firefox-ios/blob/33b7720370df304cfdb14eaae8e5b1f26cbfbcdd/Client/Frontend/Browser/BrowserViewController/BrowserViewController%2BWebViewDelegates.swift#L829) and it does not influence our outcome now that we use the same message in both cases. We can still look into that in the future and potentially even open an PR on ff's repo.

## Screnshots

Light
Simulator Screenshot - iPhone 14 Pro Max - 2023-06-27 at 15 27 38

Dark
Simulator Screenshot - iPhone 14 Pro Max - 2023-06-27 at 15 26 45

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift [hunk 1]
```diff
@@ -488,7 +491,7 @@ extension BrowserViewController: WKNavigationDelegate {
         // gives us the exact same behaviour as Safari.
         if ["sms", "tel", "facetime", "facetime-audio"].contains(url.scheme) {
             if url.scheme == "sms" { // All the other types show a native prompt
-                showSnackbar(forExternalUrl: url, tab: tab) { isOk in
+                presentExternalLinkAlert(url) { isOk in
```
Comment (d4r1091) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift:494]: Needing `// Ecosia` comment? Not a strong opinion on this. As the function won't exist anymore 😅

**Review by d4r1091 — APPROVED** [review]
✨ noiiice!
left a small non-blocking comment

**Review by ecotopian — APPROVED** [review]
Looks great to me! Please make sure that we have all the L10n before we ship it 🚢

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift (hunk 1 above)
Comment (lucaschifino) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift:494]: I added it above the `presentExternalLinkAlert` method definition (which used to be `showSnackbar`) and thought it was already enough to give context to the places where it is used as well

**Pull request merged**